ui/app/routes/workspace/projects/project/app/build.ts

```diff
@@ -34,8 +39,15 @@ export default class BuildDetail extends Route {
   }
 
   async model(params: Params): Promise<Model> {
-    let { builds } = this.modelFor('workspace.projects.project.app') as AppRouteModel;
+    let { builds, deployments, releases } = this.modelFor('workspace.projects.project.app') as AppRouteModel;
     let buildFromAppRoute = builds.find((obj) => obj.sequence === Number(params.sequence));
+    let deploymentFromAppRoute = deployments.find((obj) => {
+      if (obj.preload && obj.preload.artifact) {
+        return obj.preload.artifact.id === buildFromAppRoute?.pushedArtifact?.id;
+      }
+      return obj.artifactId === buildFromAppRoute?.pushedArtifact?.id;
```

Comment:
I suspect `preload.artifact.id` and `artifactId` will always concur when working with the real API. Maybe something we can fix in Mirage? What do you think? Worth digging into it?

Reply:
As discussed on sync, this will be a TODO and handled later since it works as-is